app/gui2/src/util/qualifiedName.ts

```diff
@@ -5,6 +5,7 @@ declare const qualifiedNameBrand: unique symbol
 const identifierRegexPart = '(?:(?:[a-zA-Z_][0-9]*)+|[!$%&*+,-./:;<=>?@\\^|~]+)'
 const identifierRegex = new RegExp(`^${identifierRegexPart}$`)
 const qnRegex = new RegExp(`^${identifierRegexPart}(?:\\.${identifierRegexPart})*$`)
+const mainSegmentRegex = new RegExp(`^(:?${identifierRegexPart}\\.${identifierRegexPart})\\.Main`)
```

Comment:
is this a typo? `:?` -> `?:`?

Reply:
also note that `[.]` is an alternative to `\\.` that doesn't need to be double escaped

Reply:
I think this group is not needed anyway.
```suggestion
const mainSegmentRegex = new RegExp(`^${identifierRegexPart}\\.${identifierRegexPart}\\.Main`)
```

Reply:
I _believe_ the group is needed, it is used when replacing in `tryQualifiedName`. And yep, `:?` is a typo.

Reply:
hmm... in that case i'm not sure it's currently working - there *is no* first capture group so the `$1` doesn't do anything - note that `(?:)` is a non-capturing group.

also, are you sure the replacement should be `'$1'` and not the empty string?

(maybe tests would be useful here? of course, not absolutely required if they're too difficult to add)

Reply:
Yep, you’re right. We shouldn’t have `:?` here, the group should be a capturing one. We have unit tests for this function, so I’m sure it is working correctly.